index.js

```diff
@@ -20,426 +38,77 @@ module.exports = function(app) {
-      ],
-      f: function(xte, originToDest, nextPoint) {
-        return toSentence([
-          '$SKAPB', 'A', 'A', Math.abs(xte), xte > 0
-            ? 'L'
-            : 'R',
-          'M',
-          'V',
-          'V',
-          (originToDest / Math.PI * 180).toFixed(0),
-          'T',
-          '00',
-          (nextPoint.bearingTrue / Math.PI * 180).toFixed(0),
-          'T',
-          (nextPoint.bearingTrue / Math.PI * 180).toFixed(0),
-          'T'
-        ]);
-      }
-    },
-
-    /*
-      RMC - Recommended Minimum Navigation Information
-      This is one of the sentences commonly emitted by GPS units.
-
-      12
-      1         2 3       4 5        6  7   8   9    10 11|  13
-      |         | |       | |        |  |   |   |    |  | |   |
-      $--RMC,hhmmss.ss,A,llll.ll,a,yyyyy.yy,a,x.x,x.x,xxxx,x.x,a,m,*hh<CR><LF>
-      Field Number:
-      1 UTC Time
-      2 Status, V=Navigation receiver warning A=Valid
-      3 Latitude
-      4 N or S
-      5 Longitude
-      6 E or W
-      7 Speed over ground, knots
-      8 Track made good, degrees true
-      9 Date, ddmmyy
-      10 Magnetic Variation, degrees
-      11 E or W
-      12 FAA mode indicator (NMEA 2.3 and later)
-      13 Checksum
-    */
-
-    RMC: {
-      title: "RMC - GPS recommended minimum",
-      keys: [
-        'navigation.datetime', 'navigation.speedOverGround', 'navigation.courseOverGroundTrue', 'navigation.position'
-      ],
-      f: function(datetime8601, sog, cog, position) {
-        var datetime = new Date(datetime8601);
-        var hours = ('00' + datetime.getHours()).slice(-2);
-        var minutes = ('00' + datetime.getMinutes()).slice(-2);
-        var seconds = ('00' + datetime.getSeconds()).slice(-2);
-        return toSentence([
-          '$SKRMC', hours + minutes + seconds + '.020',
-          'A',
-          toNmeaDegrees(position.latitude),
-          position.latitude < 0 ? 'S' : 'N',
-          toNmeaDegrees(position.longitude),
-          position.longitude < 0 ? 'W' : 'E',
-          (sog * 1.94384).toFixed(1),
-          radsToDeg(cog).toFixed(1),
-          '0000',
-          '7.0',
-          'E'
-        ]);
-      }
-    },
-
-    /*
-      PSILTBS - Proprietary target boat speed sentence for Silva => Nexus => Garmin displays
-
-
-      0     1  2
-      |     |  |
-      $PSILTBS,XX.xx,N,*hh<CR><LF>
-      Field Number:
-      0 Target Boat speed in knots
-      1 N for knots
-      2 Checksum
-    */
-    
-    PSILTBS: {
-      title: "PSILTBS - Send target boat speed to Silva/Nexus/Garmin displays",
-      keys: [
-        'performance.targetSpeed'
-      ],
-      f: function(tbs) {
-        return toSentence([
-          '$PSILTBS',
-          (tbs * 1.94384).toFixed(1),
-          'N'
-        ]);
+    function mapToNmea(encoder, name) {
+      if ( encoder.ttl === undefined) {
```

Comment:
Did I understand this right: none of the encoders have ttl defined and it can only be used in encoders, not user defined?

Reply:
the intention was that the encoder could define it if required. I didnt find I needed to reduce the frequency on my network. I guess it could be user defined as an alternative. Things like pressure and temperature don't need to be ever 2s.